src/app/core/services/rundown-timing-context-state.service.ts

```diff
@@ -0,0 +1,187 @@
+      remainingDurationInMsForRundown: 0,
+      expectedDurationsInMsForSegments: {},
+      startOffsetsInMsFromNextCursorForSegments: {},
+    }
+  }
+
+  private onRundownChanged(rundown: Rundown): void {
+    const rundownTimingContextSubject: BehaviorSubject<RundownTimingContext> | undefined = this.getRundownTimingContextSubject(rundown.id)
+    if (!rundownTimingContextSubject) {
+      return
+    }
+    const onAirSegment: Segment | undefined = rundown.segments.find(segment => segment.isOnAir)
+    const currentEpochTime: number = Date.now()
+    const expectedDurationsInMsForSegments: Record<string, number> = this.rundownTimingService.getExpectedDurationInMsForSegments(rundown)
+    const expectedDurationInMsForRundown: number = this.rundownTimingService.getExpectedDurationInMsForRundown(rundown, expectedDurationsInMsForSegments)
+    const expectedStartEpochTimeForRundown: number = this.rundownTimingService.getExpectedStartEpochTimeForRundown(rundown, expectedDurationInMsForRundown, currentEpochTime)
+    const expectedEndEpochTimeForRundown: number = this.rundownTimingService.getExpectedEndEpochTimeForRundown(rundown, expectedDurationInMsForRundown, currentEpochTime)
+    const playedDurationInMsForOnAirPart: number = this.rundownTimingService.getPlayedDurationInMsForOnAirPart(rundown, currentEpochTime)
+    const playedDurationInMsForOnAirSegment: number = this.rundownTimingService.getPlayedDurationInMsForOnAirSegment(rundown, currentEpochTime)
+    const durationInMsSpentInOnAirSegment: number = this.rundownTimingService.getDurationInMsSpentInOnAirSegment(rundown, currentEpochTime)
+    const remainingDurationInMsForRundown: number = this.rundownTimingService.getRemainingDurationInMsForRundown(rundown, expectedDurationsInMsForSegments, playedDurationInMsForOnAirSegment)
+    const rundownTimingContext: RundownTimingContext = {
+      onAirSegmentId: onAirSegment?.id,
+      currentEpochTime,
+      expectedDurationInMsForRundown: expectedDurationInMsForRundown,
+      expectedStartEpochTimeForRundown: expectedStartEpochTimeForRundown,
+      expectedEndEpochTimeForRundown: expectedEndEpochTimeForRundown,
+      remainingDurationInMsForRundown: remainingDurationInMsForRundown,
+      playedDurationInMsForOnAirSegment,
+      durationInMsSpentInOnAirSegment,
+      playedDurationInMsForOnAirPart,
+      expectedDurationsInMsForSegments,
+      startOffsetsInMsFromNextCursorForSegments: {},
+    }
+    rundownTimingContextSubject.next(rundownTimingContext)
+    this.rundowns.set(rundown.id, rundown)
+  }
+
+  private getRundownTimingContextSubject(rundownId: string): BehaviorSubject<RundownTimingContext> | undefined {
+    const rundownTimingContextSubject: BehaviorSubject<RundownTimingContext> | undefined = this.rundownTimingContextSubjects.get(rundownId)
+    if (!rundownTimingContextSubject) {
+      return
+    }
+    const { wasRemoved } = this.removeSubjectIfHasNoObserversOrSubscriptions(rundownTimingContextSubject, rundownId)
```

Comment:
Same comment as for the other place you do this.

Reply:
I have updated it to use object indexing instead of destructuring and have added the type `boolean` to the variables.